contrib/tools/meta.md/eqform.md

```diff
@@ -209,84 +215,69 @@ endfm
-
-      --- Map Formulas Over Everything
-      sort TrueLit{Term} to TrueLit,
-      sort FalseLit{Term} to FalseLit,
-      sort TruthLit{Term} to TruthLit,
-      ---
-      sort NegTruthLit{Term} to NegTruthLit,
-      sort NoTrueLit{Term} to NoTrueLit,
-      sort NoFalseLit{Term} to NoFalseLit,
-      sort NormLit{Term} to NormLit,
-      sort Lit{Term} to Lit,
-      ---
-      sort NegTruthConj{Term} to NegTruthConj,
-      sort NoTrueConj{Term} to NoTrueConj,
-      sort NoFalseConj{Term} to NoFalseConj,
-      sort NormConj{Term} to NormConj,
-      sort Conj{Term} to Conj,
-      ---
-      sort NegTruthDisj{Term} to NegTruthDisj,
-      sort NoTrueDisj{Term} to NoTrueDisj,
-      sort NoFalseDisj{Term} to NoFalseDisj,
-      sort NormDisj{Term} to NormDisj,
-      sort Disj{Term} to Disj,
-      ---
-      sort NegTruthForm{Term} to NegTruthForm,
-      sort NoTrueForm{Term} to NoTrueForm,
-      sort NoFalseForm{Term} to NoFalseForm,
-      sort NormForm{Term} to NormForm,
-      sort Form{Term} to Form) .
-
-  vars T T' : Term . vars F F' : Form . var SUB : Substitution .
-
-  op _<<_ : Form Substitution -> [Form] .
-  ---------------------------------------
-  eq tt << SUB = tt .
-  eq ff << SUB = ff .
-
+  pr SUBSTITUTION-SET .
+  pr EQFORM-IMPL{Term} * (
+    sort TrueLit{Term}          to TrueLit,
+    sort FalseLit{Term}         to FalseLit,
+    sort PosEqLit{Term}         to PosEqLit,
+    sort NegEqLit{Term}         to NegEqLit,
+    sort EqLit{Term}            to EqLit,
+    sort PosEqConj{Term}        to PosEqConj,
+    sort NegEqConj{Term}        to NegEqConj,
+    sort EqConj{Term}           to EqConj,
+    sort PosEqDisj{Term}        to PosEqDisj,
+    sort NegEqDisj{Term}        to NegEqDisj,
+    sort EqDisj{Term}           to EqDisj,
+    sort PosEqForm{Term}        to PosEqForm,
+    sort NegEqForm{Term}        to NegEqForm,
+    sort EqForm{Term}           to EqForm,
+    sort TrivTrueForm{Term}     to TrivTrueForm,
+    sort TrivFalseForm{Term}    to TrivFalseForm,
+    sort NonTrivForm{Term}      to NonTrivForm,
+    sort NoTrueForm{Term}       to NoTrueForm,
+    sort NoFalseForm{Term}      to NoFalseForm,
+    sort NormLit{Term}          to NormLit,
+    sort NormForm{Term}         to NormForm,
+    sort Form{Term}             to Form,
+    sort Eq+TrueLit{Term}       to Eq+TrueLit,
+    sort Eq+FalseLit{Term}      to Eq+FalseLit,
+    sort Eq+TrivTrueForm{Term}  to Eq+TrivTrueForm,
+    sort Eq+TrivFalseForm{Term} to Eq+TrivFalseForm,
+    sort NonTrivLit{Term}       to NonTrivLit,
+    sort NoFalseLit{Term}       to NoFalseLit,
+    sort NoTrueLit{Term}        to NoTrueLit,
+    sort Lit{Term}              to Lit) .
+
+  var F : Form .
+  vars CF CG : NoTrueForm .
+  vars DF DG : NoFalseForm .
+  vars T T' : Term .
+  var SUB : Substitution .
+
+  op _<<_ : Form Substitution -> Form .
+  eq (CF /\ CG) << SUB = (CF << SUB) /\ (CG << SUB) .
+  eq (DF \/ DG) << SUB = (DF << SUB) \/ (DG << SUB) .
```

Comment:
Let's test this to make sure it works in the cases we expect, if we want to make this change.

The original equations:

```
 ceq (F /\ F') << SUB = (F << SUB) /\ (F' << SUB) if F =/= tt /\ F' =/= tt .
 ceq (F \/ F') << SUB = (F << SUB) \/ (F' << SUB) if F =/= ff /\ F' =/= ff .
```

I think are more likely to not get caught in "sort weirdness" down the road as the sort structure shifts.

Reply:
I added tests in commit f8fce4f.